stdlib/public/stubs/Stubs.cpp

```diff
@@ -487,6 +504,11 @@ static const char *_swift_stdlib_strtoX_clocale_impl(
     const char * nptr, T* outResult, T huge,
     T (*posixImpl)(const char *, char **, locale_t)
 ) {
+  if (swift_stringIsSignalingNaN(nptr)) {
+    *outResult = std::numeric_limits<T>::signaling_NaN();
```

Comment:
In the long run we'll probably want to control the actual signaling NaN returned here so that the bit pattern of the significand is `b01xx ... xx`, but there's not a very tidy way to do that in C++ (which doesn't have anything like the Swift nan-with-payload initializer) without constructing from an explicit bit pattern. Can you please add the following TODO here to track this?
~~~~
// TODO: ensure that the returned sNaN bit pattern matches that of sNaNs produced by Swift.
~~~~